test/CQRS/LeanCode.CQRS.AspNetCore.Tests/ServiceProviderRegistrationExtensionsTests.cs

```diff
@@ -0,0 +1,161 @@
+        sp.Should().HaveService<RoleRegistry>();
+        sp.Should().HaveService<IHasPermissions>();
+        sp.Should().HaveService<ICommandValidatorResolver>();
+    }
+
+    [Fact]
+    public void Registers_handlers()
+    {
+        var sp = BuildProvider();
+
+        sp.Should().HaveService<Local.LocalCommandHandler>();
+        sp.Should().HaveService<Local.LocalQueryHandler>();
+        sp.Should().HaveService<Local.RemoteQueryHandler>();
+    }
+
+    [Fact]
+    public void Does_not_register_local_executors_by_default()
+    {
+        var sp = BuildProvider();
+
+        sp.Should().NotHaveService<ILocalCommandExecutor>();
+        sp.Should().NotHaveService<ILocalQueryExecutor>();
+        sp.Should().NotHaveService<ILocalOperationExecutor>();
+    }
+
+    [Fact]
+    public void Registers_local_executors_on_request()
+    {
+        var sp = BuildProvider(
+            c => c.WithLocalCommands(_ => { }).WithLocalQueries(_ => { }).WithLocalOperations(_ => { })
+        );
+
+        sp.Should().HaveService<ILocalCommandExecutor>();
+        sp.Should().HaveService<ILocalQueryExecutor>();
+        sp.Should().HaveService<ILocalOperationExecutor>();
+    }
+
+    [Fact]
+    public void Registers_keyed_local_executors_on_request()
+    {
+        var sp = BuildProvider(
+            c =>
+                c.WithKeyedLocalCommands("commands", _ => { })
+                    .WithKeyedLocalQueries("queries", _ => { })
+                    .WithKeyedLocalOperations("operations", _ => { })
+        );
+
+        sp.Should().HaveKeyedService<ILocalCommandExecutor>("commands");
+        sp.Should().HaveKeyedService<ILocalQueryExecutor>("queries");
+        sp.Should().HaveKeyedService<ILocalOperationExecutor>("operations");
+    }
+
+    [Fact]
+    public void Replaces_serializer_service()
+    {
+        var serializer = new CustomSerializer();
+        var sp = BuildProvider(c => c.WithSerializer(serializer));
+
+        sp.GetRequiredService<ISerializer>().Should().BeSameAs(serializer);
+    }
+
+    private static ServiceProvider BuildProvider(Action<CQRSServicesBuilder>? configure = null)
+    {
+        var collection = new ServiceCollection();
+        var builder = collection.AddCQRS(ThisCatalog, ThisCatalog);
+        configure?.Invoke(builder);
+        return collection.BuildServiceProvider();
+    }
+
+    internal class CustomSerializer : ISerializer
+    {
+        public ValueTask<object?> DeserializeAsync(
+            Stream utf8Json,
+            Type returnType,
+            CancellationToken cancellationToken
+        ) => throw new NotImplementedException();
+
+        public Task SerializeAsync(
+            Stream utf8Json,
+            object value,
+            Type inputType,
+            CancellationToken cancellationToken
+        ) => throw new NotImplementedException();
+    }
+}
+
+file class ServiceProviderAssertions : ReferenceTypeAssertions<ServiceProvider, ServiceProviderAssertions>
```

Comment:
That's something I see used for the first time.

Reply:
This is the first time where I find this useful. ;)